src/Avalonia.Controls/Presenters/ContentPresenter.cs

```diff
@@ -340,94 +338,121 @@ protected virtual IControl CreateChild()
         /// <inheritdoc/>
         protected override Size MeasureOverride(Size availableSize)
         {
-            var child = Child;
-            var padding = Padding + new Thickness(BorderThickness);
+            return Border.MeasureOverrideImpl(availableSize, Child, Padding, BorderThickness);
```

Comment:
Any particular reason that `MeasureOverrideImpl` is defined on `Border` but `ArrangeOverrideImpl` is defined on `ContentPresenter`? I feel that it would be more consistent if both were defined on `ContentPresenter`.

Reply:
Yeah, the reason is that the `MeasureOverride` algorithm applies to `Border` and `ContentPresenter` and the `ArrangeOverride` algorithm applies to `ContentPresenter` and `ScrollContentPresenter`. To me `Border` is conceptually a "lower" level control than `ContentPresenter`, so `Border` calling out to `ContentPresenter` feels wrong. Does that make sense?

Reply:
Makes sense.